lib/src/configuration.dart

```diff
@@ -311,6 +316,7 @@ class FlexibleSyncConfiguration extends Configuration {
     super.encryptionKey,
     this.syncErrorHandler = defaultSyncErrorHandler,
     this.syncClientResetErrorHandler = const ManualSyncClientResetHandler(_defaultSyncClientResetHandler),
+    this.initialSubscriptionsConfiguration,
```

Comment:
So now we have configuration inside configuration. I think this is counter intuitive. 

I suggest the following alternative

```dart
FlexibleSyncConfiguration._(
.....
{
 initialSubscriptionCallback = ....,
 reRunInitialSubscriptionCallback = false
....
}
```

so flexibleSyncConfig now has a callback and a boolen flag to rerun